### PR TITLE
Switch to loading all lights from JSON, with support for hot-swapping

### PR DESCRIPTION
**Pull request by aHooder**
This PR converts the `lights.txt` config file, and the `NpcLight`, `ObjectLight` and `ProjectileLight` enums into a single combined `lights.json` file. In addition, the JSON file is continuously watched through the file system, and changes are reflected live in-game. Switching to JSON also paves the way to more easily implement dev tools for testing different lighting configurations and making modifications on-the-fly.

https://user-images.githubusercontent.com/831317/156871145-c942c7bb-f899-40ab-9c3c-cd508d2ddebc.mp4

List of changes:
- New `ExportLightsToJson` class to convert from the old formats to JSON.
- Old enums and `lights.txt` are kept in the same locations temporarily to help with merging potential PRs that rely on old formats.
- The old `lights.txt` format parser has been moved out of the main source set. `lights.txt` itself and old enums remain for convenience with porting older PRs.
- The new `lights.json` file is now watched for changes through the file system and automatically reloaded when modified.
- Hot-swapping can also be enabled when running the plugin normally by setting the environment variable `RLHD_LIGHTS_PATH` to the path of an alternate JSON file to load. The path can be either absolute or relative to the RuneLite process' current working directory.

### CODE REVIEW

**Comment by aHooder** [comment]
I'm sneaking in shader hot-swapping as well if that's okay. There was also an issue with lists of lights being used while reloading that *should* be fixed now.

**Comment by sosodev** [comment]
Seems like this still has race conditions. It immediately crashed for me with
```
java.util.ConcurrentModificationException: null
	at java.base/java.util.ArrayList$Itr.checkForComodification(ArrayList.java:1012)
	at java.base/java.util.ArrayList$Itr.next(ArrayList.java:966)
	at rs117.hd.lighting.LightManager.update(LightManager.java:354)
	at rs117.hd.HdPlugin.drawScene(HdPlugin.java:1249)
	at gn.fw(gn.java:4719)
	at jt.gn(jt.java:4471)
	at ka.ii(ka.java:9860)
	at ka.ii(ka.java:9891)
	at ds.ij(ds.java:9749)
	at client.xt(client.java:4198)
	at client.ap(client.java:1036)
	at al.j(al.java:370)
	at al.run(al.java:324)
	at java.base/java.lang.Thread.run(Thread.java:832)
error_game_crash
```

File: src/main/resources/rs117/hd/lighting/lights.json [hunk 1]
```diff
@@ -19532,7 +19532,7 @@
       1.0,
       0.12752977
     ],
-    "type": "FLICKER",
+    "type": "PULSE",
```
Comment (sosodev) [inline src/main/resources/rs117/hd/lighting/lights.json:19535]: I assume you didn't mean to include this change to the ghost's lighting

Comment (aHooder) [reply src/main/resources/rs117/hd/lighting/lights.json:19535]: Oops, I did not! Thanks for checking.

**Comment by aHooder** [comment]
I guess I'll have to look a bit closer at preventing concurrent modifications. I'm thinking it might be easier to schedule light reloads for when the update function is called, to tie it to the render loop.

**Comment by sosodev** [comment]
> I guess I'll have to look a bit closer at preventing concurrent modifications. I'm thinking it might be easier to schedule light reloads for when the update function is called, to tie it to the render loop.

That's a great idea. :)

**Comment by aHooder** [comment]
Oops... Hold on

**Comment by sosodev** [comment]
Amazing work here Hooder. This dramatically improves the dev cycle of both lighting and general shader work. I just spent time improving the lighting in camdozaal ([before](https://cdn.discordapp.com/attachments/886739974555316294/949919838560723014/camdozaal-old.png)) ([after](https://cdn.discordapp.com/attachments/886739974555316294/949919839022108672/camdozaal-fresh.png)) and it was a rather pleasant experience. I think we can write a general guide that should make this rather accessible to noobies now. :)

**Pull request merged**